packages/react-dom/src/client/ReactDOMRoot.js

```diff
@@ -129,6 +132,10 @@ function createRootImpl(
         ? container
         : container.ownerDocument;
     eagerlyTrapReplayableEvents(container, doc);
+  } else if (enableModernEventSystem) {
+    // We use ensureListeningTo because the container might
+    // be a COMMENT_NODE.
```

Comment:
What does this have to do with comment nodes? I'm not following. The test case doesn't use them so I'm not seeing the connection.

Reply:
I've update the condition above. The comment is redundant now.